### DISCUSSION TITLE
DFRobot mmWave SEN0395 serial connection

### DISCUSSION THREAD

**Discussion by DarthWeber** [arendst/Tasmota#19794] [Q&A] [answered]
Hi, can anybody give me some hints on using the SEN0395 on an EP1 board? I can get the digital responses as switch signals but i have no success with the serial connection to adjust the SEN0395 sensor.

I set GPIO13 and GPIO14 to "Serial TX" resp. "RX" and also swapped it - but there are no responses in the tasmota console. 

Thanks,
Tom

**Comment by sfromis** [comment]
Well, do not expect any response without first doing anything on serial, like sending commands, as Tasmota has no specific support for the sensor. And better use hardware serial pins instead. I've been playing a bit with SEN0395 on ESP32 using Berry, but was disappointed in how impractical the sensor is, only providing a single detected or not output.

**Reply by DarthWeber** [reply]
Since i bought the "Everything presence one" i have no choice, the pins are fixed to GPIO 13/14 at a ESP32-WROOM-32E. But i got no response even after Sending commands.

Maybe i need to use SSerialSend instead of SerialSend? But the SSerial... commands are unknown and i didn't find out how to activate them in platformio

**Reply by sfromis** [reply, marked as answer]
With ESP32, any pins can be hardware serial, so this should be no issue. And with ESP32, I'd recommend using Berry [serial](https://tasmota.github.io/docs/Berry/#serial-class) for a device driver, instead of attempting to use commands.

Anyway, for a serial device and commands you should config the gpios as serial bridge `SerBr Tx/Rx`, not `Serial Tx/Rx`; then you should have chances of seeing activity.

**Reply by DarthWeber** [reply]
Thanks, but also with SerTX/RX i dont see any activity. But i am confused, since the commands SSerialSend are still unknown. 

Do you have a Berry example where i can see the communication?

**Reply by sfromis** [reply]
If `SSerialSend` is unknown, you did not change the config correctly.

**Reply by DarthWeber** [reply]
weird. I included "#define USE_SERIAL_BRIDGE" in my override - what else do i need to do in the config?

**Reply by sfromis** [reply]
I did not mean your build, but gpio config change.

**Comment by DarthWeber** [comment]
Oh, it's speaking now. Needed to swap RX/TX ... Thanks!

**Comment by DarthWeber** [comment]
still weird: I can communicate with the sensor, but i never see those $JYBSS,par1,par2,par3,par4* messages which should be reveived every second.

I expected them after the first communication even with a dummy string like:

`11:20:42.406 CMD: Sserialsend 1
11:20:42.415 MQT: stat/tasmota_DACE6C/RESULT = {"SSerialSend":"Done"}`

but those $JYBSS never got in.

**Reply by sfromis** [reply]
You saw it "speaking now", what was that then?

When configuring with `SerBr Tx/Rx`, I'm seeing
`{"SSerialReceived":"$JYBSS,0, , , *\r\n"}`
`{"SSerialReceived":"$JYBSS,1, , , *\r\n"}`
Maybe that's due to me already having configured something earlier.

**Reply by DarthWeber** [reply]
i mixed RX and TX after changing to SerBr .... 

Do you have some Rule on Boot?

**Reply by DarthWeber** [reply]
Ah, now i got it. I did a factory reset of the sen0395 - and it works now.
